### DISCUSSION TITLE
Override the behavior of store controller

### DISCUSSION THREAD

**Discussion by TalesFromTheNetworkRoom** [statamic-rad-pack/runway#477] [Help] [answered]
### Description

Good morning,
I've been looking for a way to modify how my model's data are stored when creating a new item in CP. (ex. I add new item through the CP's form, then some logic I wrote in the controller's override calls an API to add an element, that can't be seen nor modified from the form in CP, to the new item and save it in the DB)
Is there a part of the documentation that I should look for that would help me? 
Thanks in advance.

### Environment

Environment
Application Name: #####
Laravel Version: 10.47.0
PHP Version: 8.3.4
Composer Version: 2.2.6
Environment: local
Debug Mode: ENABLED
URL: #####
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 5
Antlers: runtime
Sites: 1
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.57.1 Solo

Statamic Addons
aryehraber/statamic-logbook: 3.2.0
mitydigital/sitemapamic: 2.3.9
plugrbase/statamic-maintenance-mode: 1.07
statamic-rad-pack/runway: 6.6.5
webographen/statamic-admin-log: 1.0.8

**Comment by duncanmcclean** [comment]
What are you looking to achieve by overriding Runway's controllers?

**Comment by TalesFromTheNetworkRoom** [comment]
Thank you for your attention,
what I need to do is to be able to modify the content that is being created during the store process.
I have an external API that I need to call and then save the response inside a column in my db. This column cannot be manually edited by users. Is there a way that I can add some logic to the the store function to be able to contact my API and then save the new item with the added response?
Thanks.

**Reply by duncanmcclean** [reply, marked as answer]
Could you use a [model observer](https://laravel.com/docs/master/eloquent#observers) to do that?

**Reply by TalesFromTheNetworkRoom** [reply]
Interesting, I will have a look at that and then notify you if your solution worked for me.
Thank you for your time.

**Reply by TalesFromTheNetworkRoom** [reply]
Good morning,
your solution worked perfectly.
Thanks again

**Reply by duncanmcclean** [reply]
Awesome, glad to hear it! 🥳
